src/TranslatorSymfonyTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace atk4\core;
+
+use InvalidArgumentException;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+trait TranslatorSymfonyTrait
+{
+    /** @var TranslatorInterface */
+    protected $translator;
+
+    /**
+     * @param TranslatorInterface $translator
+     *
+     * @return TranslatorInterface
+     * @throws Exception
+     */
+    public function setTranslator($translator)
+    {
+        if ($translator instanceof TranslatorInterface) {
```

Comment:
So Symfony/contracts will be a hard-dependency then?

Reply:
Has you see : src/TranslatorSymfonyTrait.php
is the Trait helper to use Symfony, if you want to use it you are free to use it.

every "bridged" translation class must pass from here :
```php
public function _($message, ?array $parameters = NULL, ?string $domain = NULL, ?string $locale = NULL): string
```

Reply:
Well yes, I understand that, I think I understand it a bit better now that I looked a bit at the tests.
So basically you're creating an app and adding the translator TranslatorSymfonyTrait to it, otherwise (and by default) using the TranslateableTrait and it's minimal handling, correct?
But does that support the syntax that Symfony supports? This to be exact:
```php
'{0} There are no apples|]0,1[ There is almost one apple|{1} There is one apple|[1,Inf] There is more than one apple'
```
Your code simply explodes the text on `|` and assumes that each item is the next one in the counter, yes?

Reply:
take a look here : 
[https://github.com/abbadon1334/core/blob/wip-translation/tests/TranslatableTraitTest.php]
and here : 
[https://github.com/abbadon1334/core/blob/wip-translation/tests/TranslatorSymfonyTraitTest.php]

TranslatorSymfonyTrait => is only a helper to implement method `_` in App, setTranslator and  nothing more, to be used with a standard Symfony\Translation object

I'm working on rewrite the php library of I18next.com, actual implementation in php really sucks and doesn't support all the features, is used in many js project but support really easy even nested plural and substitution, the most interesting features is that accepts interpolation of token with array objects which can be interesting in atk4/data implementation

Reply:
Ah, right.

Well from what I've seen their listed implementation is really basic, so... It actually looks a lot like my translator, which is fine for things that I'm using it for.

Still, you're just gonna use Symphony for translations for atk, right?

Reply:
From the last discussion we want to use an internal translation system but really basic, just to cover the few needs of actual atk4 translations, and leave any other complex implementation out of atk4.

i think this is the cases which at now will be covered, interpolation and simple plurals: 
https://github.com/abbadon1334/core/blob/e0c538c6179444e478cf711ae4baa1aa84147f99/tests/TranslatableTraitTest.php#L40-L53